RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -15,17 +15,19 @@ import RevenueCat
 import StoreKit
 import SwiftUI
 
-@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
 final class PurchaseHandler: ObservableObject {
 
     typealias PurchaseBlock = @Sendable (Package) async throws -> PurchaseResultData
     typealias RestoreBlock = @Sendable () async throws -> CustomerInfo
+    typealias TrackEventBlock = @Sendable (PaywallEvent) async -> Void
```

Comment:
These are now only needed for testing purposes, right? 
Is it worth keeping them around for that? Having them also forces us to have fatalError workarounds in the loading view IIRC?

Would it suffice to have a purchases mock (spy) and pass that in, so that we can directly call the methods from purchases? Seems like we only use them to check that we're calling the right methods with the right params

Reply:
These serve several purposes:
- Abstract implementation from `Purchases`
- Previews: we can have custom handlers that don't do anything or that complete a successful purchase
- Snapshots: same thing
- Unit tests (`PaywallFooterTests` for example)
- `LoadingPaywallView`: this one doesn't use real blocks for obvious reasons

> Having them also forces us to have fatalError workarounds in the loading view IIRC?

We have a `fatalError` in case you try to purchase from the loading view (which is `.disabled`, so it's impossible). Other than that, the `trackEvent` one is a no-op now.

Hopefully that answers your question why we can't just use a `Purchases` instance. We could create a `PurchasesType` implementation, but that would require implementing dozens of methods instead of just 3.

Reply:
Yeah, that's what I was getting at, essentially for DI. 

I don't feel strong enough about this to continue to block this PR, but I do think there are alternative approaches worth studying, that wouldn't necessitate building a mock of all methods in `PurchasesType` from scratch. 

I worry about us adding a different approach for testing (instead of DI of a class, DI at the closure level). It makes things more complex for a brand new maintainer. And I'd argue that the closures are harder to read and wrap your head around than just passing in a purchases object, although that bit is subjective. 

Alternate approaches that I think would also do the trick: 
- find a way to share the purchases mock instance we use for other tests. We already share some tests code, this wouldn't fall too far from that
- create a new, much smaller protocol, that only has the methods that we'd need (purchase, restore, track events). Refactor PurchasesType so that it also conforms to this protocol. Then we pass in an instance of this smaller protocol, which is easy to mock. Then PurchasesHandler just takes a `purchases: NewSmallType = Purchases.shared` as param, and the rest of the class is dead simple.

Reply:
Thanks for clarifying what you meant! It does indeed lead to a simpler implementation :)

Done in #3196